Rocket.Core/IOC/DependencyContainer.cs

```diff
@@ -179,16 +113,38 @@ public bool IsRegistered(Type type, string mappingName = null)
             return container.IsRegistered(type, mappingName);
         }
 
-        public T TryGet<T>(string mappingName = null)
+        public bool TryGet<T>(string mappingName, out T output)
         {
-            GuardRegistered(typeof(T), mappingName, false);
-            return container.Resolve<T>(mappingName, new OrderedParametersOverride(new object[0]));
+            try
+            {
+                GuardRegistered(typeof(T), mappingName);
+                output = container.Resolve<T>(mappingName, new OrderedParametersOverride(new object[0]));
+
+                return true;
+            }
+            catch
+            {
+                output = default(T);
```

Comment:
Please don't use exceptions for control flow. Making the Try* methods act like .NET's TryCast is a good idea, i'd propose we drop the GuardRegistered method at all then, and simply check if it exists directly - no exceptions involved.

I had the guards in place because I expect functionality to completely break when an unavailable dependency is required with TryGet(), having to abort the flow in that case. In that scenario a plain Get() can be null. Fetching a dependency with TryGet will NEVER be null, the policy is "Comply or die".

If we change the behaviour of TryGet to be allow the developer to decide what to do, and leave him worry about what 
 exception to throw now we don't have any benefit over fetching the dependency with Try and checking if null. I am OK with having TryGet behave like eg. TryCast, but in such scenarios we HAVE to throw exceptions if the dependency is not found in Get()